frontend/src/components/ServerContentDisplayAndEditing.tsx

```diff
@@ -0,0 +1,397 @@
+import {Ups} from "../types/ups.ts";
+import {useEffect, useState} from "react";
+import axios from "axios";
+import {useNavigate} from "react-router-dom";
+import {Server} from "../types/server.ts";
+import {Credentials} from "../types/credentials.ts";
+import CredentialsSelect from "./CredentialsSelect.tsx";
+import UpsSelect from "./UpsSelect.tsx";
+import FormBottom from "./FormBottom.tsx";
+import NameAndAddressInputFields from "./NameAndAddressInputFields.tsx";
+
+type EditProps = {
+    server: Server,
+    upses: Ups[],
+    credentialsList: Credentials[],
+    serverUpdate: () => void
+}
+
+
+export default function UpsContentDisplayAndEditing(props: Readonly<EditProps>) {
+
+    const [server, setServer] = useState<Server>({
+        id: "new",
+        name: "",
+        address: "",
+        upsId: "",
+        shutdownTime: 180,
+        credentials: {id: "", user: "", password: "", global: false}
+    })
+    const [editing, setEditing] = useState<boolean>(false)
+    const [changedData, setChangedData] = useState<boolean>(false)
+    const [nameInput, setNameInput] = useState<string>("")
+    const [addressInput, setAddressInput] = useState<string>("")
+    const [upsSelection, setUpsSelection] = useState<string>("")
+    const [changedCredentialsInput, setChangedCredentialsInput] = useState<boolean>(false)
+    const [localSelected, setLocalSelected] = useState<boolean>(true)
+    const [userInput, setUserInput] = useState<string>("")
+    const [passwordInput, setPasswordInput] = useState<string>("")
+    const [globalCredentialsSelection, setGlobalCredentialsSelection] = useState<string>("")
+    const [shutdownSecondsInput, setShutdownSecondsInput] = useState<number>(3)
+    const [message, setMessage] = useState<string>("")          // in case of errors and for warning before deletion
+    const confirmationMessage: string = "Really delete? Reclick button to confirm."
+    const navigate = useNavigate()
+
+    function switchEditMode(state: boolean): void {
+        setEditing(state)
+        setChangedData(false)
+        setChangedCredentialsInput(false)
+        setMessage("")
+    }
+
+    function backToList(updated: boolean): void {
+        switchEditMode(false)
+        if (updated) props.serverUpdate()
+        navigate("/server")
+    }
+
+    /** initialize data from props */
+    useEffect(() => {
+        setServer(props.server)
+        setEditing(props.server.id === "new")
+        setNameInput(props.server.name)
+        setAddressInput(props.server.address)
+        if (props.server.credentials === null) {
+            setGlobalCredentialsSelection("")
+            setLocalSelected(true)
```

Comment:
Isn't line 64-66 redundant as those are the default values?

Reply:
I want to prevent that data from previously displayed objects remain in the state.